bionic/deriver.py

```diff
@@ -684,9 +680,13 @@ def _compute_result_group_for_dnode(self, dnode):
 
 
 @attr.s(frozen=True)
-class Bootstrap:
+class ExecutionCore:
```

Comment:
What do you think of `CoreEntities`? We use the term `core entities` in some places but not `execution`?

Reply:
`ExecutionCore` seems fine.

I think my concern is more about consistency in naming and comments.

Reply:
`CoreEntities` is not a bad name but I prefer to avoid using plurals in class names, since a variable called `_core_entities` would look like a collection. On the other hand, it's hard to think of a singular noun that makes sense here. "Core" was the best one I could think of, and I added "Execution" because that's what it's used for. (We don't use the term "execution" consistently but so far I think it's the best term for what happens in `flow_execution.py` and `task_execution.py`.) Note also that not every built-in entity appears in the `ExecutionCore`, and one of the core fields (`task_key_logging_level`) is not an entity at all.

Our use of "core" and "internal" is indeed a bit confusing. I think we currently tend to use "internal entity" to describe any built-in entity, but they were originally named as `core__...` and it's hard to change user-facing names, so we haven't done that (yet).

After writing all this, given that we're stuck with the user-facing term "core" for the time being, I wonder if re-using the term in our code is unnecessarily confusing. A similar term would be "nucleus" -- it sounds a bit weird but at least it doesn't have any other connotations. Other possibilities could be "kernel" or "infrastructure", but they're both suggestive of other concepts.

What do you all think?

Reply:
I tried renaming "core" to "nucleus" but ended up liking it less -- passing around a "nucleus" feels weirder and less intuitive to me than a "core". For now I'm leaving it as "core", but I'm open to revisiting it.